### PR TITLE
[QnA] Add QnAMakerRecognizer, QnAMakerRecognizerTests and fix issues

### PR DESCRIPTION
**Pull request by Batta32**
Fixes #

## Specific Changes

  - 
  -
  -

### CODE REVIEW

File: libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/QnAMakerRecognizer.java [hunk 1]
```diff
@@ -0,0 +1,552 @@
...
+                        recognizerResult.getEntities() != null
+                                ? Serialization.toStringSilent(recognizerResult.getEntities())
+                                : null);
+                put("AdditionalProperties",
+                        !recognizerResult.getProperties().isEmpty()
+                                ? Serialization.toStringSilent(recognizerResult.getProperties())
+                                : null);
+            }
+        };
+
```
Comment (fran893) [inline libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/QnAMakerRecognizer.java:540]: Here is missing this [part](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder.AI.QnA/QnAMakerRecognizer.cs#L329), is that correct?

File: libraries/bot-ai-qna/src/test/java/com/microsoft/bot/ai/qna/QnAMakerRecognizerTests.java [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    public void logPiiIsFalseByDefault() {
+        QnAMakerRecognizer recognizer = new QnAMakerRecognizer() {
+            {
+                setHostName(hostname);
+                setEndpointKey(endpointKey);
+                setKnowledgeBaseId(knowledgeBaseId);
+            }
+        };
+        Activity activity = MessageFactory.text("hi");
+        TurnContext context = new TurnContextImpl(new TestAdapter(), activity);
```
Comment (fran893) [inline libraries/bot-ai-qna/src/test/java/com/microsoft/bot/ai/qna/QnAMakerRecognizerTests.java:54]: `context` var is not being used

File: libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/QnAMakerRecognizer.java [hunk 3]
```diff
@@ -0,0 +1,552 @@
...
+    @JsonProperty("qnaId")
+    private Integer qnAId = 0;
+
+    @JsonProperty("logPersonalInformation")
+    private Boolean logPersonalInformation = false;
+
+    /**
+     * Initializes a new instance of the {@link QnAMakerRecognizer} class.
+     */
+    public QnAMakerRecognizer() {
```
Comment (FedericoBernal) [inline libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/QnAMakerRecognizer.java:88]: Why is this empty? if its being called won't it break the execution? and if it's not being called, shouldn't we delete it?

**Review by fran893 — CHANGES_REQUESTED** [review]
The methods names are missing the qnAMakerRecognizer prefix at the beginning. This is for all the tests methods. Also, the order of the methods are not the same as in C#.

File: libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/QnAMakerRecognizer.java (hunk 3 above)
Comment (Batta32) [reply libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/QnAMakerRecognizer.java:88]: Agree with you, I think that we can remove it.

I'll do it :)

File: libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/QnAMakerRecognizer.java [hunk 4]
```diff
@@ -0,0 +1,552 @@
...
+            });
+        } */
+
+        // if there is $qna.metadata set add to filters
+        Metadata[] externalMetadata = this.metadata;
+        if (externalMetadata != null) {
+            filters.addAll(Arrays.asList(externalMetadata));
+        }
+
+        QnAMakerOptions options = new QnAMakerOptions() {
```
[the remark this answers is not in the document]
Comment (Batta32) [reply libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/QnAMakerRecognizer.java:403]: I moved it outside to be more legible

File: libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/QnAMakerRecognizer.java (hunk 1 above)
Comment (Batta32) [reply libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/QnAMakerRecognizer.java:540]: +1. I'll update this

File: libraries/bot-ai-qna/src/test/java/com/microsoft/bot/ai/qna/QnAMakerRecognizerTests.java (hunk 2 above)
Comment (Batta32) [reply libraries/bot-ai-qna/src/test/java/com/microsoft/bot/ai/qna/QnAMakerRecognizerTests.java]: Removed

**Comment by Batta32** [comment]
@fran893 & @FedericoBernal , I applied the requested changes.

Related to your [comment](https://github.com/southworks/botbuilder-java/pull/53#pullrequestreview-592161027), as mentioned in the SUP, this was taken from JS instead of C#.

**Comment by fran893** [comment]
Related to this [comment](https://github.com/southworks/botbuilder-java/pull/53#issuecomment-780559318), the other unit tests were migrated from C# and now we have inconsistency between out unit tests. Probably this could be fixed later

**Review by fran893 — APPROVED** [review]

**Comment by Batta32** [comment]
As soon as AdaptiveDialogs is added, we can migrate the unit tests of C# correctly.

**Review by FedericoBernal — APPROVED** [review]
LGTM

**Pull request merged**